src/fides/api/models/privacy_request.py

```diff
@@ -937,19 +926,46 @@ def pause_processing_for_email_send(self, db: Session) -> None:
         self.status = PrivacyRequestStatus.awaiting_email_send
         self.save(db=db)
 
+    def get_request_task_celery_task_ids(self) -> List[str]:
+        """Returns the celery task ids for each of the Request Tasks (subtasks)
+
+        It is possible Request Tasks get queued multiple times, so the celery task
+        id returned is the last celery task queued.
+        """
+        request_task_celery_ids: List[str] = []
+        for request_task in self.request_tasks:
+            request_task_id: Optional[str] = request_task.get_cached_task_id()
+            if request_task_id:
+                request_task_celery_ids.append(request_task_id)
+        return request_task_celery_ids
+
     def cancel_processing(self, db: Session, cancel_reason: Optional[str]) -> None:
-        """Cancels a privacy request.  Currently should only cancel 'pending' tasks"""
+        """Cancels a privacy request.  Currently should only cancel 'pending' tasks
+
+        Just in case, also tries to cancel sub tasks (Request Tasks) if applicable,
+        although these shouldn't exist if the Privacy Request is pending.
+        """
         if self.canceled_at is None:
             self.status = PrivacyRequestStatus.canceled
             self.cancel_reason = cancel_reason
             self.canceled_at = datetime.utcnow()
             self.save(db)
 
-            task_id = self.get_cached_task_id()
-            if task_id:
-                logger.info("Revoking task {} for request {}", task_id, self.id)
-                # Only revokes if execution is not already in progress
-                celery_app.control.revoke(task_id, terminate=False)
+            task_ids: List[
+                str
+            ] = (
+                self.get_request_task_celery_task_ids()
+            )  # Celery tasks for sub tasks (DSR 3.0 Request Tasks)
+            parent_task_id = (
+                self.get_cached_task_id()
+            )  # Celery task for current Privacy Request
+            if parent_task_id:
+                task_ids.append(parent_task_id)
+
+            for celery_task_id in task_ids:
+                logger.info("Revoking task {} for request {}", celery_task_id, self.id)
+                # Only revokes if execution is not already in progress.
+                celery_app.control.revoke(celery_task_id, terminate=False)
```

Comment:
The model-level method to cancel privacy requests also tries to revoke the celery task - I've added trying to revoke the associated celery tasks for the Request Task as well. Further, Request Task execution has a new check early on that fails early if the overall Privacy Request was moved to "canceled".

This isn't super likely to get hit as we currently only allow pending Privacy Requests to be cancelled, perhaps because it was tricky to interrupt the Privacy Request running in memory in a reliable way.  We'd probably have more flexibility now to interrupt this with DSR 3.0 if desired.